src/normalizers/keywordTranslatorMaker.js

```diff
@@ -0,0 +1,31 @@
+import { regexMatchingDecoderMaker, regexMatchingEncoderMaker } from './regex.js';
+
+export function keywordTranslatorMaker (name, keywordMap) {
+    if (keywordMap) {
+        const encoder = regexMatchingEncoderMaker(
+            name,
+            /(?<protector>protector:\w+)/g,
+            Object.fromEntries(Object.keys(keywordMap).map(kw => [ `protector:${kw}`, keywordMap[kw] ])),
+            findReplacement
+        );
+        const decoder = regexMatchingDecoderMaker(
+            name,
+            new RegExp(`(?<protector>${Object.keys(keywordMap).join("|")||"NO_PROTECTED_STRINGS"})`, 'g'),
+            (groups) => ({ t: 'x', v: `protector:${groups.protector}` }),
+            findReplacement
+        );
+        return [ decoder, encoder ]    
+    } else {
+        throw 'You have to specify a keyword map as in input paramter';
+    }
+}
+
+//Checks for values in the map based on the flags
+function findFlagValue (charMap, flags) {
```

Comment:
this recursive approach against random values found in flags seems very error-prone. The flags object is shared and we have no idea what kind of values it may contain. we have more control over the keys so i think we should be explicit about what specific keys to inspect into the flags object.

i'm not sure how strong the requirement to have multiple keys is (i don't see any tests with nested keys). maybe the built-in default behavior is to expect the keywordMap to contain an object with the key matching the value of `flags.targetLang` (which is a reliable value because it's system-generated and not config-depended). if people need arbitrary complex logic, then we can allow to pass to keywordTranslatorMaker an optional extra callback parameter and when present we can pass in the matched element in the `keywordMap` and just use whatever the callback returns.

Reply:
I'm not fully convinced about the nested keys either.  I removed the recursive search for now.  I didn't add the extra callback yet.